### PR TITLE
Move main step logic to an exported functon.

### PR DESCRIPTION
**Pull request by lpusok**
- This allows reuse of step logic in other projects
- Introduce new error type to represent a top level step error. This allows custom error handling for analytics for example.

### CODE REVIEW

File: activatesshkey/util.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+func splitEnv(env string) (string, string) {
+	e := strings.Split(env, "=")
+	return e[0], strings.Join(e[1:], "=")
+}
+
+func ensureSavePath(savePath string) error {
+	dirpath := filepath.Dir(savePath)
+	return os.MkdirAll(dirpath, 0700)
```
Comment (godrei) [inline activatesshkey/util.go:41]: in the previous version, we used different permission: [0600](https://github.com/bitrise-steplib/steps-activate-ssh-key/blob/master/main.go#L69)
is this change intentional?

File: main.go [hunk 2]
```diff
@@ -33,172 +19,11 @@ func main() {
 
 	log.SetEnableDebugLog(cfg.Verbose)
```
Comment (godrei) [inline main.go:20]: will this turn on debug logging in `activatesshkey` package too?
(previously debug logging applied for the functions moved to the `activatesshkey` package)

**Review by godrei — CHANGES_REQUESTED** [review]
I left a few questions.
Note: bitrise.yml could be updated (remove old step share wf)

File: activatesshkey/util.go (hunk 1 above)
Comment (lpusok) [reply activatesshkey/util.go:41]: Yes, if the directory did not exist previously, will not have permission to create files inside with the existing permissions.

File: main.go (hunk 2 above)
Comment (lpusok) [reply main.go:20]: It does as this option is set globally in the log package. This will mean that verbose logging will be enabled also in the project scanner step.

**Review by godrei — APPROVED** [review]
